### PR TITLE
Delete unused mono/benchmark folder

### LINKED ISSUE #47628: GPLv2 licensed file in source code
In the repository there is file logic.cs which is licensed on GPLv2 (runtime/src/mono/mono/benchmark/logic.cs). 

Doesn't that pose legal complications to anyone distributing this library? Does this file end up in the final dlls ?

### PR DESCRIPTION
**Pull request by akoeplinger**
Fixes https://github.com/dotnet/runtime/issues/47628

### CODE REVIEW

File: src/mono/mono/benchmark/Makefile [hunk 1]
```diff
@@ -0,0 +1 @@
+# empty placeholder for netcore build
```
Comment (stephentoub) [inline src/mono/mono/benchmark/Makefile:1]: Why do we need the folder here at all?

Comment (akoeplinger) [reply src/mono/mono/benchmark/Makefile:1]: Yeah it's a quirk of how autotools works. We'll get rid of it once we turn off the mirror soon.

**Review by naricc — APPROVED** [review]

**Comment by CoffeeFlux** [comment]
Do we know that we have an equivalent to these in BDN? Maybe worth comparing to see if there's anything here that would be useful to port over?

Otherwise, glad to see it cleaned up.

**Comment by akoeplinger** [comment]
As far as I know we're not running these benchmarks on the mono side either (they're very old), but @naricc can confirm whether there's anything interesting to salvage.

**Comment by naricc** [comment]
Just going by the names and assuming they are reasonably descriptive, it looks like most of these are effectively covered by something already in the dotnet/performance suite, although I am not sure about zipmark.

As far as a I know no one had tried to port these to use BDN.

**Review by SamMonoRT — APPROVED** [review]

**Pull request merged**